pkg/analysis/nodurations/analyzer.go

```diff
@@ -0,0 +1,142 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package nodurations
+
+import (
+	"fmt"
+	"go/ast"
+
+	"golang.org/x/tools/go/analysis"
+	"golang.org/x/tools/go/analysis/passes/inspect"
+	"golang.org/x/tools/go/ast/inspector"
+	kalerrors "sigs.k8s.io/kube-api-linter/pkg/analysis/errors"
+	"sigs.k8s.io/kube-api-linter/pkg/analysis/utils"
+)
+
+const name = "nodurations"
+
+// Analyzer is the analyzer for the nodurations package.
+// It checks that no struct field is of a type either time.Duration or metav1.Duration.
+var Analyzer = &analysis.Analyzer{
+	Name:     name,
+	Doc:      "Use seconds instead of Duration to avoid need to implement go duration parsing",
```

Comment:
```suggestion
	Doc:      "Duration types should not be used, to avoid need to for clients to implement go duration parsing. Instead, use integer based fields with the unit in the field name.",
```